clustermesh/clustermesh.go

```diff
@@ -127,7 +127,8 @@ func (k *K8sClusterMesh) generateService() *corev1.Service {
 			svc.Spec.Type = corev1.ServiceTypeLoadBalancer
 			svc.ObjectMeta.Annotations["service.beta.kubernetes.io/aws-load-balancer-internal"] = "0.0.0.0/0"
 		default:
-			svc.Spec.Type = corev1.ServiceTypeClusterIP
+			svc.Spec.Type = corev1.ServiceTypeNodePort
```

Comment:
I'm on the fence here. NodePort makes it work but it's super fragile as it will depend on the node IPs to stay around forever. I would feel better if we error out and require the type to be set to NodePort specifically with an explanation of what type of assumption the user enters in this case.

Reply:
Noticed this when updating the Cilium GSG for External Workloads, resolved there by placing a warning in the docks.

Under what circumstances can the (previous) default of ClusterIP be used to access the service from outside of the cluster? Only if direct routing is used? Or do you mean that if we (would) default to ClusterIP service type we actually error out with instructions on how `--service-type` `NodePort` can be used for testing with these caveats?

Reply:
OK, made this to fail out with instructions if `--service-type` is not explicitly set and none of the LoadBalancer options can be auto-detected. Also added warnings if service type NodePort is used.